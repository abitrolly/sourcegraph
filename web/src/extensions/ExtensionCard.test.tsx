import React from 'react'
import { PlatformContext } from '../../../shared/src/platform/context'
import { ExtensionCard } from './ExtensionCard'
import { mount } from 'enzyme'

describe('ExtensionCard', () => {
    const NOOP_PLATFORM_CONTEXT: PlatformContext = {} as any

    test('renders', () => {
        expect(
            mount(
                <ExtensionCard
                    node={{
                        id: 'x/y',
                        manifest: {
                            activationEvents: ['*'],
                            description: 'd',
                            url: 'https://example.com',
                            icon: 'data:image/png,abcd',
                        },
                        registryExtension: {
                            id: 'abcd1234',
                            extensionIDWithoutRegistry: 'x/y',
                            url: 'extensions/x/y',
                            isWorkInProgress: false,
                            viewerCanAdminister: false,
                        },
                    }}
                    subject={{ id: 'u', viewerCanAdminister: false }}
                    settingsCascade={{ final: null, subjects: null }}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
        ).toMatchSnapshot()
    })
})
