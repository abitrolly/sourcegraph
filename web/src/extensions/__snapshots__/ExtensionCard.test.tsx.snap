// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionCard renders 1`] = `
<ExtensionCard
  node={
    Object {
      "id": "x/y",
      "manifest": Object {
        "activationEvents": Array [
          "*",
        ],
        "description": "d",
        "icon": "data:image/png,abcd",
        "url": "https://example.com",
      },
      "registryExtension": Object {
        "extensionIDWithoutRegistry": "x/y",
        "id": "abcd1234",
        "isWorkInProgress": false,
        "url": "extensions/x/y",
        "viewerCanAdminister": false,
      },
    }
  }
  platformContext={Object {}}
  settingsCascade={
    Object {
      "final": null,
      "subjects": null,
    }
  }
  subject={
    Object {
      "id": "u",
      "viewerCanAdminister": false,
    }
  }
>
  <div
    className="d-flex"
  >
    <div
      className="extension-card card"
    >
      <div
        className="card-body extension-card__body d-flex flex-column position-relative"
        onClick={[Function]}
      >
        <div
          className="d-flex"
        >
          <img
            className="extension-card__icon mr-2"
            src="data:image/png,abcd"
          />
          <div
            className="text-truncate w-100"
          >
            <div
              className="d-flex align-items-center"
            >
              <h4
                className="card-title extension-card__body-title mb-0 mr-1 text-truncate font-weight-normal flex-1"
              >
                <LinkOrSpan
                  className="stretched-link"
                  to="extensions/x/y"
                >
                  <AnchorLink
                    className="stretched-link"
                    to="extensions/x/y"
                  >
                    <a
                      className="stretched-link"
                      href="extensions/x/y"
                    >
                      <Path
                        path="x/y"
                      >
                        <span
                          className="text-muted"
                        >
                          x
                          /
                        </span>
                        <strong>
                          y
                        </strong>
                      </Path>
                    </a>
                  </AnchorLink>
                </LinkOrSpan>
              </h4>
              <ExtensionConfigurationState
                className="small"
                enabledIconOnly={true}
                isAdded={false}
                isEnabled={false}
              >
                <span
                  className="text-muted small"
                >
                  Disabled
                </span>
              </ExtensionConfigurationState>
            </div>
            <div
              className="mt-1"
            >
              <div
                className="text-muted text-truncate"
              >
                d
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</ExtensionCard>
`;
